### PR TITLE
Publish connector development best practices guide

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Publish best practices and checklist that will be used to "certify" connectors

### CODE REVIEW

File: docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Best Practices for Connector Development
+
+In order to guarantee the highest quality for connectors, we've compiled the following best practices for connector development. Connectors which follow these best practices will be labelled as "Airbyte Certified" to indicate they've passed a high quality bar and will perform reliably in all production use cases. Following these guidelines is **not required** for your contribution to Airbyte to be accepted, as they add a barrier to entry for contribution.
```
Comment (michel-tricot) [inline docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md:3]: I would add the same mention at the end (`but it doesn't hurt`)

File: docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#### Integration Testing
+
+**As much as possible, prove functionality via testing**. This means slightly different things depending on the type of connector: 
+
+* **All connectors** must test all the sync modes they support during integration tests
+* **Database connectors** should test that they can replicate **all** supported data types in both `read` and `discover` operations
+* **API connectors** should validate records that every stream outputs data
+  * If this causes rate limiting problems, there should be a periodic CI build which tests this on a less frequent cadence to avoid rate limiting
+
+**Thoroughly test edge cases.** While Airbyte provides a [Standard Test Suite](testing-connectors.md#airbytes-standard-tests) that all connectors must pass, it's not possible for the standard test suite to cover all edge cases. When in doubt about whether the standard tests provide sufficient evidence of functionality, write a custom test case for your connector.
```
Comment (michel-tricot) [inline docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md:34]: we will need a tutorial to write a custom test case.

**Review by michel-tricot — APPROVED** [review]

File: docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md (hunk 2 above)
Comment (sherifnada) [reply docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md]: created issue

File: docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md (hunk 1 above)
Comment (sherifnada) [reply docs/contributing-to-airbyte/building-new-connector/best-practices-for-connector-development.md]: done

**Pull request merged**